lib/src/view/game/game_result_dialog.dart

```diff
@@ -98,6 +99,62 @@ class _GameEndDialogState extends ConsumerState<GameResultDialog> {
           padding: const EdgeInsets.only(bottom: 16.0),
           child: GameResult(game: gameState.game),
         ),
+        AnimatedSize(
```

Comment:
Instead of using `AnimatedSize` with `Visibility` have you tried `AnimatedCrossfade`? This is higher level and the result may be better since it animates size and opacity.
```dart
AnimatedCrossFade(
  duration: const Duration(milliseconds: 400),
  firstChild: Sizedbox.shrink(),
  secondChild: RematchOfferDialog(...),
  crossFadeState: gameState.game.opponent?.offeringRematch == true ? CrossFadeState.showSecond : CrossFadeState.showFirst,
)
```